### PR TITLE
Add mapping key identification to format

### LINKED ISSUE #35: Marking mapping key locations
(This suggestion is not original to me, it comes originally from someone at Nomic Labs.)  A useful thing to include would be to somehow mark which SHA3 instructions are for mapping keys.

Just filing this as a quick issue rather than adding this into what exists, hope that's OK!

### PR DESCRIPTION
**Pull request by haltman-at**
A quick stab at what mapping key information might look like in the format.  Intended to be open to future expansion.  Figured for different mapping hash formats, it makes more sense to just have an enumeration (that can be expanded in the future) rather than trying to specify them in some sort of DSL. :P  There's not just one out there, Solidity and Vyper do things a bit differently here!

(Not sure if "prefix" and "postfix" are intuitive enough, arguably those should be swapped, or they could just be renamed to something less ambiguous.)

I'm assuming this is only used on SHA3 instructions.  Potentially one might want to allow some way to mark other sorts of hashes were those ever used for this purpose, but, I figure, there's no need to go addressing that unless someone actually does it, which seems unlikely.

Closes #35.

### CODE REVIEW

File: docs/source/format.md [hunk 1]
```diff
@@ -238,6 +244,7 @@ that contributed to the generation of this opcode.
 * `ast`: A list of AST ids for the "closest" AST node that contributed to the generation of this opcode.
 * `stack` A layout of the stack as understood by the compiler, represented as a list.
 * `jumps`: If present, provides hints about the location being jumped to by a jumping command (JUMP or JUMPI)
+* `mappings`: If preent, contains information about how the opcode relates to mapping keys.
```
Comment (ekpyron) [inline docs/source/format.md:247]:
```suggestion
* `mappings`: If present, contains information about how the opcode relates to mapping keys.
```

File: docs/source/format.md [hunk 2]
```diff
@@ -321,3 +328,10 @@ If the value of `sort` is `"return"`, then the dictionary has the following fiel
 * `returns`: A list of dictionaries with the same format of as the `arguments` array of `call`, but without any `return_address` entries.
 
 **Discussion**: The above proposal doesn't really handle the case of "tail-calls" identified at the beginning of this document, where multiple return addresses can be pushed onto the stack. Is that something debug format must explicitly model?
+
+#### Mapping key identification
+
+The value of this field (when present) is a dictionary with (some of) the following fields:
+* `isMappingHash`: A boolean that identifies whether the opcode is computing a hash for a mapping.
```
Comment (cameel) [inline docs/source/format.md:335]: This is a big vague, but I'm assuming only a single opcode that returns the final result should be marked? I.e. you're not interested in the whole sequence of opcodes that are involved in a high-level operation of indexing a mapping, only the final step giving you the hash, right?

Comment (haltman-at) [reply docs/source/format.md:335]: Yes, that's what I meant, sorry, likely that should be clarified.

Of course, there's one big exception to that statement, which is the case of mapping computations that themselves involve a hash at an earlier step (as can happen in Vyper).  In that case the earlier SHA3 instruction should also be marked, but as a prehash instead.

**Review by gnidan — COMMENTED** [review]
@jtoman you bring up a good point that we can generalize this

**Review by gnidan — APPROVED** [review]
But consensus says we can merge this now!

**Pull request merged**